.github/PULL_REQUEST_TEMPLATE/manual_maintenance.md

```diff
@@ -37,6 +37,9 @@ Update versions in the following dockerfiles
 - [ ] `./dataland-frontend/DockerfileTest`
 - [ ] `./dataland-internal-storage/Dockerfile`
 - [ ] `./dataland-internal-storage/DockerfileTest`
+- [ ] `./dataland-qa-service/Dockerfile`
```

Comment:
Not relevant here - but: We need to figure out if our consumers are idempotent - i.e. that everything is fine, if a message is received twice.

Reply:
How would that happen? For that the sender would have to send the message more than once, right? Because once a consumer read a message the message is deleted from the queue.

Reply:
Yes, but if you consider the consumer to be an independent microservice of the producer, you could argue that the consumer should be resilient against whatever might appear on the exchange. I.e. you could have some kind of message verification on the consumer side if the consumer really wants to process a message it receives.

Since all services are part of the same repository and developed by the same team I'd argue that we can rely on the content of the message and their number for now and do not need to verify the content of messages received. 

From a testing perspecitve, an e2e API test might be able to somewhat cover this. When a new dataset is being uploaded, after a while, only a single new QA'd dataset should have been produced.

Reply:
Yes, i will look into this

Reply:
Currently we use the default mode, which is automatic acknowledgement, which in the documentation is descripbed as: "In automatic acknowledgement mode, a message is considered to be successfully delivered immediately after it is sent." (https://www.rabbitmq.com/confirms.html#acknowledgement-modes)

Therefore we will need to introduce manual acknowledgement to archieve "delivered at least once"